cartridge_account/src/lib.cairo

```diff
@@ -213,11 +204,28 @@ mod Account {
             self._set_public_key(_public_key);
         }
 
-        fn validate_transaction(self: @ContractState) -> felt252 {
+        fn validate_transaction(self: @ContractState, mut calls: Array<Call>) -> felt252 {
+            let tx_info = get_tx_info().unbox();
+            let tx_hash = tx_info.transaction_hash;
+            let mut signature = tx_info.signature;
+            if signature.len() == 2_u32 {
+                return self.validate_ecdsa_transaction();
+            } 
+            let signature_type = *signature.at(0_u32);
+            if signature_type == 'Session Token v1' {
+                SessionImpl::validate_session_serialized(self, self.get_public_key(), signature, calls.span())
+            } else if signature_type == 'Webauthn v1' {
+                WebauthnImpl::verifyWebauthnSignerSerialized(self, signature, tx_hash)
```

Comment:
Would be great to have the signature types in an enum or constants.

Reply:
Done :)